cruise-control-client/cruisecontrolclient/client/cccli.py

```diff
@@ -0,0 +1,353 @@
+#!/usr/bin/env python3
+
+# To be able to easily parse command-line arguments
+import argparse
+
+# To be able to make more precise type hints
+from typing import Dict, Tuple, Type  # noqa
+
+# Convenience functions for displaying the retrieved Response
+from cruisecontrolclient.client.Display import display_response
+
+# To be able to instantiate Endpoint objects
+import cruisecontrolclient.client.Endpoint as Endpoint
+
+# To be able to make requests and get a response given a URL to cruise-control
+from cruisecontrolclient.client.Responder import JSONDisplayingResponderGet, JSONDisplayingResponderPost
+
+# To be able to compose a URL to hand to requests
+from cruisecontrolclient.client.Query import generate_url_from_cc_socket_address
+
+
+def add_add_parameter_argument(p: argparse.ArgumentParser):
+    """
+    This should be used with all cruise-control endpoint parsers, to provide
+    forward compatibility and greater operational flexibility.
+    :param p:
+    :return:
+    """
+    p.add_argument('--add-parameter', '--add-parameters', metavar='PARAM=VALUE',
+                   help="Manually specify one or more parameter and its value in the cruise-control endpoint, like 'param=value'",
+                   nargs='+')
+
+
+def add_remove_parameter_argument(p: argparse.ArgumentParser):
+    """
+    Adds the ability to manually specify parameters to remove from the cruise-control
+    endpoint, of the form 'parameter'.
+
+    This should be used with all cruise-control endpoint parsers, to provide
+    forward compatibility and greater operational flexibility.
+
+    :param p:
+    :return:
+    """
+    p.add_argument('--remove-parameter', '--remove-parameters', metavar='PARAM',
+                   help="Manually remove one or more parameter from the cruise-control endpoint, like 'param'",
+                   nargs='+')
+
+
+# Define the argparse flags that cannot be used for endpoint parameters
+#
+# This helps the multi-stage argument parsing not to conflict with itself
+# during the different stages.
+non_parameter_flags = {
+    'add_parameter',
+    'endpoint_subparser',
+    'socket_address',
+    'remove_parameter',
+}
+
+# Define the in-order available endpoints for programmatically building the argparse CLI
+available_endpoints: Tuple[Type[Endpoint.AbstractEndpoint]] = (
+    Endpoint.AddBrokerEndpoint,
+    Endpoint.AdminEndpoint,
+    Endpoint.BootstrapEndpoint,
+    Endpoint.DemoteBrokerEndpoint,
+    Endpoint.FixOfflineReplicasEndpoint,
+    Endpoint.KafkaClusterStateEndpoint,
+    Endpoint.LoadEndpoint,
+    Endpoint.PartitionLoadEndpoint,
+    Endpoint.PauseSamplingEndpoint,
+    Endpoint.ProposalsEndpoint,
+    Endpoint.RebalanceEndpoint,
+    Endpoint.RemoveBrokerEndpoint,
+    Endpoint.ResumeSamplingEndpoint,
+    Endpoint.StateEndpoint,
+    Endpoint.StopProposalExecutionEndpoint,
+    Endpoint.TrainEndpoint,
+    Endpoint.UserTasksEndpoint
```

Comment:
The following 3 endpoints seems to be missing from this tuple:
```
1. REVIEW_BOARD
2. REVIEW
3. TOPIC_CONFIGURATION
```
* Please see `com.linkedin.kafka.cruisecontrol.servlet.EndPoint`

Reply:
These are endpoints that have been added to cruise-control since the cruise-control-client was implemented.

It is of course possible to add these endpoints to the client, but that will require more than trivial changes to the proposed RB.

Reply:
I think it is ok to add support for these endpoints on a separate patch. In that case, to remember our plan, is it possible to create an issue for it?

Reply:
https://github.com/linkedin/cruise-control/issues/784